From 72d0766592c4017bd975047236e3727ec5399325 Mon Sep 17 00:00:00 2001
From: Kentaro Hayashi <hayashi@clear-code.com>
Date: Tue, 4 Mar 2025 18:10:27 +0900
Subject: [PATCH 3/7] Omit shm lock support

Android does not have /dev/shm support, so disable it

Signed-off-by: Kentaro Hayashi <hayashi@clear-code.com>
---
 libpod/lock/shm/shm_lock.go           | 2 +-
 libpod/lock/shm_lock_manager_linux.go | 2 +-
 libpod/runtime.go                     | 4 +++-
 3 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/libpod/lock/shm/shm_lock.go b/libpod/lock/shm/shm_lock.go
index 24031a8cab..a3c568f146 100644
--- a/libpod/lock/shm/shm_lock.go
+++ b/libpod/lock/shm/shm_lock.go
@@ -1,4 +1,4 @@
-//go:build (linux || freebsd) && cgo
+//go:build ((linux && !android) || freebsd) && cgo
 
 package shm
 
diff --git a/libpod/lock/shm_lock_manager_linux.go b/libpod/lock/shm_lock_manager_linux.go
index ba438b9c61..6a1dd61bf6 100644
--- a/libpod/lock/shm_lock_manager_linux.go
+++ b/libpod/lock/shm_lock_manager_linux.go
@@ -1,4 +1,4 @@
-//go:build linux
+//go:build linux && !android
 
 package lock
 
diff --git a/libpod/runtime.go b/libpod/runtime.go
index fc358de026..4d8cb40801 100644
--- a/libpod/runtime.go
+++ b/libpod/runtime.go
@@ -13,7 +13,7 @@ import (
 	"slices"
 	"strings"
 	"sync"
-	"syscall"
+	_ "syscall"
 	"time"
 
 	"github.com/containers/buildah/pkg/parse"
@@ -252,6 +252,7 @@ func getLockManager(runtime *Runtime) (lock.Manager, error) {
 			}
 		}
 
+	/*
 	case "", "shm":
 		lockPath := define.DefaultSHMLockPath
 		if rootless.IsRootless() {
@@ -284,6 +285,7 @@ func getLockManager(runtime *Runtime) (lock.Manager, error) {
 				return nil, err
 			}
 		}
+	*/
 	default:
 		return nil, fmt.Errorf("unknown lock type %s: %w", runtime.config.Engine.LockType, define.ErrInvalidArg)
 	}
-- 
2.49.0

